spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb

```diff
@@ -0,0 +1,49 @@
+RSpec.describe Api::V0::SubscriptionsController, type: :request do
+  describe 'POST /api/v0/subscriptions' do
+    let(:university) { create(:user, role: :university) }
+    let(:credentials) { university.create_new_auth_token }
+    let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(credentials) }
+    before do
+      StripeMock.start
+      post '/api/v0/subscriptions',
+           headers: headers,
+           params: { stripeToken: StripeMock.generate_card_token }
+    end
+    after { StripeMock.stop }
+
+    it 'return a 200 status' do
+      expect(response.status).to eq 200
+    end
+
+    it 'returns a success message' do
+      expect(response_json['message']).to eq 'Payment successful'
+    end
+
+    it 'requires a charge amount' do
+      expect {
+        charge =
+          Stripe::Charge.create(
+            currency: 'sek', customer: university, stripeToken: :stripeToken
+          )
+      }.to raise_error(Stripe::InvalidRequestError)
```

Comment:
Why are you raising the `Stripe::InvalidRequestError` ?
While in the happy path you are raising ` 'Payment successful'
`